docs/aws/audit/esmonitoring/rules/es_ri_recent_purchases_remediation.mdx

```diff
@@ -1,6 +1,251 @@
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the Elasticsearch service. 
+
+2. In the Elasticsearch dashboard, look for the "Reserved Instances" section. This is where you can see all the purchased reserved instances for Elasticsearch.
+
+3. Review the list of purchased reserved instances. Check the "Purchase Date" column to identify recent purchases. 
+
+4. Evaluate the recent purchases based on your organization's usage patterns and needs. Check if the instance type, size, and term match your requirements. If there are any discrepancies or unusual purchases, it might indicate a misconfiguration or an oversight.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List all Elasticsearch domains: Use the following AWS CLI command to list all the Elasticsearch domains in your AWS account.
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domains in your AWS account.
+
+3. Describe Elasticsearch domain: For each domain listed in the previous step, use the following AWS CLI command to get detailed information about the domain.
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <DomainName>
+   ```
+   Replace `<DomainName>` with the name of the Elasticsearch domain. This command will return detailed information about the specified Elasticsearch domain, including its configuration.
+
+4. Review Reserved Instance purchases: In the output of the previous command, look for the `ReservedElasticsearchInstanceOfferingId` field. This field indicates the ID of the Reserved Instance offering that was used to purchase the domain. If this field is present and not empty, it means that a Reserved Instance was recently purchased for the domain. You should review this purchase to ensure that it was intentional and is cost-effective.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+from datetime import datetime, timedelta
+```
+
+2. Create a session using your AWS credentials. Replace 'your_access_key', 'your_secret_access_key', and 'your_region' with your actual AWS credentials and region.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='your_access_key',
+    aws_secret_access_key='your_secret_access_key',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>